### ISSUE TITLE
Difficulties reproducing validation results with the test pipeline

### ISSUE THREAD

**Issue by NanBlanc** [Pointcept/PointTransformerV3#146]
I trained on a custom dataset, using as reference for config `s3dis/semseg-pt-v3m1-0-base.py`
I tried only one epoch, to have validate the whole pipeline.

here is the exact config file :
```
_base_ = ["../_base_/default_runtime.py"]

# misc custom setting
batch_size = 2  # bs: total bs in all gpus
num_worker = 24
mix_prob = 0.8
empty_cache = False
enable_amp = True

# model settings
model = dict(
    type="DefaultSegmentorV2",
    num_classes=4,
    backbone_out_channels=64,
    backbone=dict(
        type="PT-v3m1",
        in_channels=4,
        order=("z", "z-trans", "hilbert", "hilbert-trans"),
        stride=(2, 2, 2, 2),
        enc_depths=(2, 2, 2, 6, 2),
        enc_channels=(32, 64, 128, 256, 512),
        enc_num_head=(2, 4, 8, 16, 32),
        enc_patch_size=(1024, 1024, 1024, 1024, 1024),
        dec_depths=(2, 2, 2, 2),
        dec_channels=(64, 64, 128, 256),
        dec_num_head=(4, 4, 8, 16),
        dec_patch_size=(1024, 1024, 1024, 1024),
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        attn_drop=0.0,
        proj_drop=0.0,
        drop_path=0.3,
        shuffle_orders=True,
        pre_norm=True,
        enable_rpe=False,
        enable_flash=True,
        upcast_attention=False,
        upcast_softmax=False,
        cls_mode=False,
        pdnorm_bn=False,
        pdnorm_ln=False,
        pdnorm_decouple=True,
        pdnorm_adaptive=False,
        pdnorm_affine=True,
        pdnorm_conditions=("ScanNet", "S3DIS", "Structured3D"),
    ),
    criteria=[
        dict(type="CrossEntropyLoss", weight=[0.01,0.45,0.45,0.09], loss_weight=1.0, ignore_index=-1),
        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, per_image=True, ignore_index=-1),
    ],
)

# scheduler settings
epoch = 2
eval_epoch=1
optimizer = dict(type="AdamW", lr=0.006, weight_decay=0.05)
scheduler = dict(
    type="OneCycleLR",
    max_lr=[0.006, 0.0006],
    pct_start=0.05,
    anneal_strategy="cos",
    div_factor=10.0,
    final_div_factor=1000.0,
)
param_dicts = [dict(keyword="block", lr=0.0006)]

# dataset settings
dataset_type = "QCSFDataset"
data_root = "/home/ostocker/travaux/data/compact"

data = dict(
    num_classes=4,
    ignore_index=4,
    names=[
        "ground",
        "abies balsamea",
        "picea mariana",
        "low vegetation",
    ],
    train=dict(
        type=dataset_type,
        split="train",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="RandomDropout", dropout_ratio=0.2, dropout_application_ratio=0.2
            ),
            dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="x", p=0.5),
            dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="y", p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),

            dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            dict(type="SphereCrop", sample_rate=0.6, mode="random"),
            dict(type="SphereCrop", point_max=204800, mode="random"),
            dict(type="CenterShift", apply_z=False),
            
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
    ),
    val=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
    ),
    test=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
        ],
        test_mode=True,
        test_cfg=dict(
            voxelize=dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="test",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            crop=None,
            post_transform=[
                dict(type="CenterShift", apply_z=False),
                dict(type="ToTensor"),
                dict(
                    type="Collect",
                    keys=("coord", "grid_coord", "segment"),
                    feat_keys=("coord", "strength"),
                ),
            ],
            aug_transform=[],
        ),
    ),
)
```
My custom dataset is pretty simple : 
```
@DATASETS.register_module()
class QCSFDataset(DefaultDataset):
    def __init__(self, ignore_index=-1, **kwargs):
        self.intensity_max=1025
        super().__init__(ignore_index=ignore_index, **kwargs)
        
    def get_data_list(self):
        data_path=os.path.join(self.data_root,self.split)
        data_list=ost.getFileBySubstr(data_path,"ply")
        print("=> FOUND",len(data_list),"files in",data_path)
        return data_list
        
    def get_data(self, idx):
        data_path = self.data_list[idx % len(self.data_list)]
        data, fields = ost.readPly(data_path,fields_names=True)
        data=data.astype(np.float32)
        coord = data[:, :3]
        strength = data[:, 3].reshape([-1, 1]) / self.intensity_max  # scale intensity to [0, 1]
        
        if "class" in fields:
            segment = data[:, fields.index("class")].astype(np.int32)
        else:
            segment = np.ones((data.shape[0],), dtype=np.int64) * self.ignore_index

        data_dict = dict(
            coord=coord,
            strength=strength,
            segment=segment,
            name=self.get_data_name(idx),
        )
        #print(data_dict)
        return data_dict
```
When i train i got these validation results in the log, on the **validation set** :
```
[2025-03-18 22:48:13,953 INFO evaluator.py line 174 932088] Val result: mIoU/mAcc/allAcc 0.3743/0.5447/0.5320.
[2025-03-18 22:48:13,953 INFO evaluator.py line 180 932088] Class_0-ground Result: iou/accuracy 0.6904/0.7349
[2025-03-18 22:48:13,953 INFO evaluator.py line 180 932088] Class_1-abies balsamea Result: iou/accuracy 0.1217/0.1229
[2025-03-18 22:48:13,953 INFO evaluator.py line 180 932088] Class_2-picea mariana Result: iou/accuracy 0.3608/0.9948
[2025-03-18 22:48:13,953 INFO evaluator.py line 180 932088] Class_3-low vegetation Result: iou/accuracy 0.3244/0.3262
```
Which are promising results, but when running the test pipeline on the **test set** it failed completely : 
```
[2025-03-18 22:48:27,704 INFO test.py line 332 932088] Val result: mIoU/mAcc/allAcc 0.0534/0.2500/0.2138
[2025-03-18 22:48:27,704 INFO test.py line 338 932088] Class_0 - ground Result: iou/accuracy 0.2138/1.0000
[2025-03-18 22:48:27,704 INFO test.py line 338 932088] Class_1 - abies balsamea Result: iou/accuracy 0.0000/0.0000
[2025-03-18 22:48:27,704 INFO test.py line 338 932088] Class_2 - picea mariana Result: iou/accuracy 0.0000/0.0000
[2025-03-18 22:48:27,704 INFO test.py line 338 932088] Class_3 - low vegetation Result: iou/accuracy 0.0000/0.0000
```

Which i assumed was due too early overfitting. But, when i run the test pipeline on the **validation set** to see if i can reproduce the validation results i got this : 
```
[2025-03-26 17:09:31,214 INFO test.py line 332 1287635] Val result: mIoU/mAcc/allAcc 0.0590/0.2500/0.2360
[2025-03-26 17:09:31,214 INFO test.py line 338 1287635] Class_0 - ground Result: iou/accuracy 0.2360/1.0000
[2025-03-26 17:09:31,214 INFO test.py line 338 1287635] Class_1 - abies balsamea Result: iou/accuracy 0.0000/0.0000
[2025-03-26 17:09:31,214 INFO test.py line 338 1287635] Class_2 - picea mariana Result: iou/accuracy 0.0000/0.0000
[2025-03-26 17:09:31,214 INFO test.py line 338 1287635] Class_3 - low vegetation Result: iou/accuracy 0.0000/0.0000
```

And i have the same problem even when inputting the train set, making me believe that something is wrong with my setup of the test pipeline. Nevertheless i have the same parameters as the s3dis config for the test (expect for input feature as i use only strengh and no color or normals).
So how can it be that i can not reproduce the `evaluator.py` results on a similar dataset than the `test.py` ?

**Comment by Gofinge** [comment]
Oh, I see. Please put at least one augmentation pipeline in  `aug_transform=[],`, like a dummy augmentation: `dict(type="RandomScale", scale=[1, 1])`

**Comment by NanBlanc** [comment]
Thanks a lot, that was it.
